### PR TITLE
Updates for OLM packaging and deployment issues.

### PR DESCRIPTION
**Pull request by dgoodwin**
This is a first draft of the CSV for Hive to be delivered as an OLM package/bundle.

This can successfully be published to a Quay application repo with `operator-courier push config/operator-bundle dgoodwin hive-operator 0.0.5 $TOKEN`

Everything here is currently copied manually. The CRDs will be easy to auto-update, the CSV much more tricky as parts of it are embedded objects, but not the entire Kube Deployment/Role/RoleBinding we already have yaml for... messy.

### CODE REVIEW

File: config/operator-bundle/hive-clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,186 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: hive-operator-0.0.1
```
Comment (dgoodwin) [inline config/operator-bundle/hive-clusterserviceversion.yaml]: @jmelis any idea if we should use a static name here?

File: config/operator-bundle/hive-clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      OpenShift cluster provisioning and management at scale.
+    containerImage: quay.io/dgoodwin/hive:latest
+    createdAt: 2019-02-25T14:29:00Z
+    support: Devan Goodwin
+spec:
+  displayName: Hive
+  description: |
+    OpenShift cluster provisioning and management at scale.
+  keywords: ['kubernetes', 'openshift', 'multi-cluster', 'cluster']
+  version: 0.0.1
```
Comment (dgoodwin) [inline config/operator-bundle/hive-clusterserviceversion.yaml]: @jmelis it looks like it's valid here to do 0.0.1-HASH here. What would be easiest for your pipeline in this regard?

Maybe a make csv target that builds this file up with the version = 0.0.1-HASH, replaces the container images with a hash tag as well? Your pileline can run that, then publish the resulting bundle with operator-courier?

File: config/operator-bundle/hive-clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+          template:
+            metadata:
+              labels:
+                control-plane: hive-operator
+                controller-tools.k8s.io: "1.0"
+            spec:
+              serviceAccountName: hive-operator
+              containers:
+              # By default we will use the latest CI images published from hive master:
+              - image: registry.svc.ci.openshift.org/openshift/hive-v4.0:hive
```
Comment (dgoodwin) [inline config/operator-bundle/hive-clusterserviceversion.yaml]: This would need replacing as well.

File: Makefile [hunk 4]
```diff
@@ -75,7 +75,7 @@ install: crd rbac
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
 .PHONY: deploy
-deploy: generate manifests
+deploy: install generate manifests
```
Comment (csrwng) [inline Makefile]: shouldn't you call manifests before generate, since the result of manifests can affect the generated bindata.go?

**Comment by csrwng** [comment]
@dgoodwin apart from the comment above, things look good to me. It'd be nice if we had a make target and some instructions (in the README) on how to generate the csv. But that could definitely be done in a follow-up PR.

File: hack/generate-operator-bundle.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    if (os.path.isfile(os.path.join('config/crds', file_name))):
+        shutil.copy(full_path, os.path.join(outdir, file_name))
+
+with open('config/templates/hive-csv-template.yaml', 'r') as stream:
+    csv = yaml.load(stream)
+
+csv['spec']['install']['spec']['clusterPermissions'] = []
+
+# Add our operator role to the CSV:
+with open('config/operator/operator_role.yaml', 'r') as stream:
```
Comment (csrwng) [inline hack/generate-operator-bundle.py]: Just for future improvement ... wouldn't we want to add `config/operator/*_role.yaml` to the roles?

Comment (dgoodwin) [reply hack/generate-operator-bundle.py]: I don't forsee more roles for the operator, but maybe we could move this one  alongside the others in config/rbac/, and then add config/rbac/*_role.yaml.

**Comment by dgoodwin** [comment]
Updated. I will def add some docs on how to gen the CSV, process might evolve quite a bit but hopefully soon.

**Comment by dgoodwin** [comment]
Sorry @csrwng :( I think this missing was why hiveadmission did not have perms to read it's configmap.

**Comment by dgoodwin** [comment]
I have expanded this PR in scope as I kept discovering problems all related to getting this out to hive-stage, which is needed ASAP.

Added commits to transition to namespace "hive", and fix service account permission problems resulting from my attempt to kubectl apply -f config/rbac, instead of letting them flow through kustomize. 

We probably want to drop kubebuilder annotations for generating rbac soon.

@csrwng sorry, needs another look. :(

**Comment by dgoodwin** [comment]
[test]

File: config/hiveadmission/hiveadmission_rbac_role.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (csrwng) [inline config/hiveadmission/hiveadmission_rbac_role.yaml]: maybe for validating later, but I don't believe hiveadmission needs any permissions for dnszones. We are able to validate other resources (clusterdeployments, clusterimagesets) without explicit permissions to those.

**Pull request merged**